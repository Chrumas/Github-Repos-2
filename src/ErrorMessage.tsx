import React from "react";
import { useEffect } from "react";
import { StyledError } from "./styles/Error.styled";
import { motion, useAnimation } from "framer-motion";

interface props {
  inputColor: string;
  inputName: string;
}

//Customizable error message
export const ErrorMessage = ({ inputColor, inputName }: props) => {
  const changeMessage = useAnimation();
  useEffect(() => {
    changeMessage.start({
      scale: [0, 1],
      opacity: [0, 1],
      transition: { duration: 0.2 },
    });
  }, [changeMessage, inputColor, inputName]);

  const style = {
    color: inputColor,
  };
  if (inputName !== "")
    return (
      <StyledError
        as={motion.div}
        exit={{ opacity: 0 }}
        animate={changeMessage}
      >
        <div data-testid="error" className="error" style={style}>
          {inputName}
        </div>
      </StyledError>
    );
  else return <></>;
};
