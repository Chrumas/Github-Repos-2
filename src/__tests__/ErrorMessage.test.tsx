import * as React from "react";
import * as ReactDOM from "react-dom";
import { ErrorMessage } from "../ErrorMessage";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ErrorMessage inputColor="red" inputName="any" />, div);
});

it("renders message correctly", () => {
  const { getByTestId } = render(
    <ErrorMessage inputColor="red" inputName="any" />
  );
  // eslint-disable-next-line testing-library/prefer-screen-queries
  expect(getByTestId("error")).toHaveTextContent("any");
});

it("renders with correct color", () => {
  const { getByTestId } = render(
    <ErrorMessage inputColor="red" inputName="any" />
  );
  // eslint-disable-next-line testing-library/prefer-screen-queries
  expect(getByTestId("error")).toHaveStyle("color: red");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<ErrorMessage inputColor="red" inputName="any" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
