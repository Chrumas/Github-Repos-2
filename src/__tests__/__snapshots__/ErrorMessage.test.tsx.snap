// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdvvtL kUelhD"
  style={Object {}}
>
  <div
    className="error"
    data-testid="error"
    style={
      Object {
        "color": "red",
      }
    }
  >
    any
  </div>
</div>
`;
